packages/tldraw/src/test/frames.test.ts

```diff
@@ -234,6 +256,58 @@ describe('frame shapes', () => {
 			h: 50,
 		})
 	})
+	it('Unparents a shape when resize causes it to be out of bounds', () => {
+		const rectId: TLShapeId = createRect({ pos: [70, 10] })
+		dragCreateFrame()
+		// resize the frame so the shape is out of bounds
+		editor.pointerDown(100, 50, { target: 'selection', handle: 'right' })
+		editor.pointerMove(50, 50)
+		editor.pointerUp(50, 50)
+		const parent = editor.getShape(rectId)?.parentId
+		expect(parent).toBe('page:page')
+	})
+
+	it('doesnt unparent a shape that is only partially out of bounds', () => {
+		const rectId: TLShapeId = createRect({ pos: [70, 10] })
+		const frameId = dragCreateFrame()
+		// resize the frame so the shape is partially out of bounds
+		editor.pointerDown(100, 50, { target: 'selection', handle: 'right' })
+		editor.pointerMove(70, 50)
+		editor.pointerUp(70, 50)
+		const parent = editor.getShape(rectId)?.parentId
+		expect(parent).toBe(frameId)
+	})
+
+	it('does not parent a shape when resizing over it', () => {
+		const rectId = createRect({ pos: [70, 10] })
+		// create frame next to shape
+		dragCreateFrame({ move: [60, 100], up: [60, 100] })
+		// resize the frame so the shape is totally covered
+		editor.pointerDown(60, 50, { target: 'selection', handle: 'right' })
+		editor.pointerMove(100, 50)
+		editor.pointerUp(100, 50)
+		const parent = editor.getShape(rectId)?.parentId
+		expect(parent).toBe('page:page')
+	})
+
+	it('Moves children when resizing a parent frame', () => {
+		const rectId: TLShapeId = createRect()
+		dragCreateFrame()
+		editor.pointerDown(0, 0, { target: 'selection', handle: 'top_left' })
```

Comment:
Would probably also do an expect on the `y` position. I generally like to make sure that things (that I will expect later on) are in the correct state before the action.